### PR TITLE
[tracking-transparency] Add permission hooks from modules factory

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Adds a permission hook from #13782 to `expo-tracking-transparency`. 

# How

Exported permission hook and permission hook options type.

# Test Plan

See example in docblock.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by byCedric** [comment]
I noticed this package doesn't have a code owner, do you want me to add you @cruzach?

File: packages/expo-tracking-transparency/src/TrackingTransparency.ts
Comment (cruzach) [inline packages/expo-tracking-transparency/src/TrackingTransparency.ts:80]: let's give this the same description as the methods: https://github.com/expo/expo/pull/13864/files#diff-ef0edece81d91e4ff128fcebc377ce08b51c4b29999acd36fa109b3447dc2be8R20-R30 

and we can take out `needsAudit` then 👍

File: packages/expo-tracking-transparency/CHANGELOG.md
Comment (cruzach) [inline packages/expo-tracking-transparency/CHANGELOG.md:9]:
```suggestion
- Add `useTrackingPermissions` hook from modules factory. ([#13864](https://github.com/expo/expo/pull/13864) by [@bycedric](https://github.com/bycedric))
```

**Review by cruzach — COMMENTED** [review]
> I noticed this package doesn't have a code owner, do you want me to add you @cruzach?

Yeah, that's good with me 👍

**Review by cruzach — APPROVED** [review]
lgtm 👍 also if you wanna add me to codeowners in this PR, that's fine. If not I can commit it separately 😄

**Pull request merged**
